### DISCUSSION TITLE
Interval for Sending 6LoWPAN Fragmented Packets in OpenThread

### DISCUSSION THREAD

**Discussion by fuxiaoming-lumi** [openthread/openthread#10578] [Q&A] [answered]
I am currently exploring the configuration options within OpenThread for managing the interval between sending 6LoWPAN fragmented packets from the sender. While I have come across parameters like **OPENTHREAD_CONFIG_IP6_FRAG_REASSEMBLY_TIMEOUT** that relate to reassembly timeouts on the receiver side. 

Does OpenThread support configuration options to adjust the interval between sending 6LoWPAN fragmented packets from the sender?

**Comment by abtink** [comment, marked as answer]
What's the background/reason for this? 

If you want to space out frame transmissions to allow the receiver to forward the previous frame onwards, you can consider the "collision avoidance delay" feature. Note that this applies to all frames (including fragmented frames), and importantly used with "lowpan mesh header" frames that would be forwarded. 

```c++
/**
 * @def OPENTHREAD_CONFIG_MAC_COLLISION_AVOIDANCE_DELAY_ENABLE
 *
 * Define as 1 to enable collision avoidance delay feature, which adds a delay wait time after a successful frame tx
 * to a neighbor which is expected to forward the frame. This delay is applied before the next direct frame tx (towards
 * any neighbor) on an FTD.
 *
 * The delay interval is specified by `OPENTHREAD_CONFIG_MAC_COLLISION_AVOIDANCE_DELAY_INTERVAL` (in milliseconds).
 *
 */
#ifndef OPENTHREAD_CONFIG_MAC_COLLISION_AVOIDANCE_DELAY_ENABLE
#define OPENTHREAD_CONFIG_MAC_COLLISION_AVOIDANCE_DELAY_ENABLE 1
#endif

/**
 * @def OPENTHREAD_CONFIG_MAC_COLLISION_AVOIDANCE_DELAY_INTERVAL
 *
 * Specifies the collision avoidance delay interval in milliseconds. This is added after a successful frame tx to a
 * neighbor that is expected to forward the frame (when `OPENTHREAD_CONFIG_MAC_COLLISION_AVOIDANCE_DELAY_ENABLE` is
 * enabled).
 *
 */
#ifndef OPENTHREAD_CONFIG_MAC_COLLISION_AVOIDANCE_DELAY_INTERVAL
#define OPENTHREAD_CONFIG_MAC_COLLISION_AVOIDANCE_DELAY_INTERVAL 8
#endif
```

**Comment by fuxiaoming-lumi** [comment]
Thank you very much. This is exactly what I wanted. @abtink 
We want to do some tests for some very low-power scenarios. When the battery is about to run out, for some large data packets, f a certain interval can be set during the packet sending process, it may save battery life.
